packages/neos-ui/src/manifest.js

```diff
@@ -247,11 +248,24 @@ manifest('main', {}, globalRegistry => {
             store.dispatch(actions.UI.PageTree.focus(parentContextPath));
         }
 
+        // If we are removing current document node...
         if ($get('ui.contentCanvas.contextPath', state) === contextPath) {
-            const parentNodeUri = $get(['cr', 'nodes', 'byContextPath', parentContextPath, 'uri'], state);
+            let redirectContextPath = contextPath;
+            let redirectUri = null;
+            // Determine closest parent that is not being removed
+            while (!redirectUri) {
+                redirectContextPath = parentNodeContextPath(redirectContextPath);
+                // This is an extreme case when even the top node does not exist in the given dimension
+                // TODO: still find a nicer way to break out of this situation
```

Comment:
We could fallback to the `siteNodeContextPath` for a start - that should be always present. But we can deal with that when we get rid of the hard-coded Uris alltogether.

Reply:
But it is not present in all dimensions... In any case the logic above would traverse up to the site node in given dimension, and if it isn't there it would rudely reload the whole thing (should be extremely rare). And yes, needs taking care when we want to get rid of hardcoded uris.